starfive/visionfive/v2/default.nix

```diff
@@ -1,12 +1,17 @@
 { config, lib, pkgs, ... }: {
   boot = {
-    # Force no ZFS (from nixos/modules/profiles/base.nix) until updated to kernel 6.0
-    supportedFilesystems =
-      lib.mkForce [ "btrfs" "reiserfs" "vfat" "f2fs" "xfs" "ntfs" "cifs" ];
     consoleLogLevel = lib.mkDefault 7;
-    kernelPackages = lib.mkDefault (pkgs.callPackage ./linux-6.6.nix {
-      inherit (config.boot) kernelPatches;
-    });
+    # Require at least Linux >= 6.9
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
+    # Enable some required flags for JH7110 arch
+    kernelPatches = [{
+      name = "enable_jh7110_arch";
+      patch = null;
+      extraStructuredConfig = with lib.kernel; {
+        PINCTRL_STARFIVE_JH7110_SYS = yes;
```

Comment:
`PINCTRL_STARFIVE_JH7110_SYS` defaults to `SOC_STARFIVE`, which is already `y` in riscv defconfig. Is our kernel build system doing something funny?

Reply:
Maybe we should do
```diff
--- a/lib/systems/platforms.nix
+++ b/lib/systems/platforms.nix
@@ -535,6 +535,7 @@ rec {
       name = "riscv-multiplatform";
       target = "Image";
       autoModules = true;
+      preferBuiltin = true;
       baseConfig = "defconfig";
       DTB = true;
       extraConfig = ''
``
same as `aarch64-multiplatform`.

Reply:
See https://github.com/NixOS/nixpkgs/pull/316724